pkg/trace/agent/agent.go

```diff
@@ -485,7 +485,8 @@ func isManualUserDrop(priority sampler.SamplingPriority, pt traceutil.ProcessedT
 }
 
 // sample reports the number of events found in pt and whether the chunk should be kept as a trace.
```

Comment:
Probably worth a comment that it does a semi-deep copy of the `*traceutil.ProcessedTrace` in this function to avoid making accidental changes to which spans are in the trace. But note that any changes made directly to the spans, such as setting tags, etc. is not allowed.